### DISCUSSION TITLE
Vulkan renderer

### DISCUSSION THREAD

**Discussion by GreyXor** [servo/servo#34561] [Ideas] [answered]
Hi,
I believe I understand that Servo currently uses [WebRender](https://github.com/servo/webrender), which relies on [OpenGL](https://github.com/servo/webrender/blob/bd7704a7091c51593cef9a3fe2ba86eb7e26359e/webrender/src/renderer/mod.rs#L390) for rendering. I’m curious to know if there are any ongoing efforts or plans to implement a Vulkan-based renderer for Servo. If so, could you share any details about the progress or challenges involved in such an implementation?

Thank you for your time!

**Comment by jdm** [comment, marked as answer]
There are no such efforts or plans.

**Reply by GreyXor** [reply]
Thank you @jdm,
Do you think it would make sense? Any opinions on this?

**Reply by jdm** [reply]
The most likely possibility is that webrender is rewritten on top of wgpu. That's still an enormous project with no timeline. Once that exists, the GL parts of Servo would need to be rewritten on top of wgpu/some platform-specific renderer as well (surfman, compositor, webxr, webgl, media). I don't see that work happening unless it's funded by a company that has a use case which cannot rely on opengl, and they have engineers to throw at the problem.

**Reply by jdm** [reply]
I guess one other possibility is if there was an opengl implementation that was backed by vulkan under the hood, which could theoretically trade off some performance to avoid having to rewrite everything. I don't know enough about the graphics landscape to say if this is realistic or not.

**Reply by GreyXor** [reply]
Thanks, that's very informative. There's [zink](https://docs.mesa3d.org/drivers/zink.html), as `opengl implementation that was backed by vulkan under the hood`.

**Comment by zyxhere** [comment]
Hmm wonder where the runtime dependency on vulkan-loaders comes from https://book.servo.org/running-servoshell.html#runtime-dependencies

**Reply by jdm** [reply]
Possibly from wgpu, which is used to implement WebGPU?
